sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/tests/TimeSeriesInsightsTypesTests.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using FluentAssertions;
+using FluentAssertions.Common;
+using NUnit.Framework;
+
+namespace Azure.IoT.TimeSeriesInsights.Tests
+{
+    public class TimeSeriesInsightsTypesTests : E2eTestBase
+    {
+        private static readonly TimeSpan s_retryDelay = TimeSpan.FromSeconds(10);
+        private const int MaxNumberOfRetries = 10;
+
+        public TimeSeriesInsightsTypesTests(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        [Test]
+        public async Task TimeSeriesInsightsTypes_Lifecycle()
+        {
+            // Arrange
+            TimeSeriesInsightsClient client = GetClient();
+            var timeSeriesTypes = new List<TimeSeriesType>();
+            var tsiTypeNamePrefix = "type";
+            int numOfTypesCreated = 0;
+            var timeSeriesTypesProperties = new Dictionary<string, string>
+            {
+                { Recording.GenerateAlphaNumericId(tsiTypeNamePrefix), Recording.GenerateId()},
+                { Recording.GenerateAlphaNumericId(tsiTypeNamePrefix), Recording.GenerateId()}
+            };
+
+            // Build aggregate type
+            var countExpression = new TimeSeriesExpression("count()");
+            var aggValue = new AggregateVariable(countExpression);
+            var variables = new Dictionary<string, TimeSeriesVariable>();
+            var variableNamePrefix = "var";
+            variables.Add(Recording.GenerateAlphaNumericId(variableNamePrefix), aggValue);
+
+            foreach (var property in timeSeriesTypesProperties)
+            {
+                var type = new TimeSeriesType(property.Key, variables);
+                type.Id = property.Value;
+                timeSeriesTypes.Add(type);
+                numOfTypesCreated++;
+            }
+
+            // Act and assert
+            try
+            {
+                // Get all Time Series types in the environment
+                AsyncPageable<TimeSeriesType> getAllTypesResponse = client.GetTimeSeriesTypesAsync();
```

Comment:
Because this is an E2E test, I don't see a create time series types before this step. When this test runs on the pipeline, the response will be empty as it's a brand new TSI environment.

Reply:
Wouldn't it still be a default Type?

Reply:
I see. Yup agreed.